package fetch.cfg.device;

// Configuration for the PI C843 Stage controller

message C843StageController
{ 
  message Axis
	{
		optional uint32 id    = 1 [default=1];         //1-4
		optional string stage = 2 [default="NOSTAGE"]; //M-511.DD (XY), M-501.DG (Z)
	}

	optional uint32 id   = 1 [default=1];
	repeated Axis   axis = 2;
}

message SimulatedStage
{
  message Axis
  { optional float min_mm = 1 [default=  0.0];
    optional float max_mm = 2 [default=100.0];
  }
	repeated Axis   axis = 2;
}

message Stage
{
  enum StageType
  {
    Simulated = 0;
    C843      = 1;
  }    
  enum TilingMode
  { PixelAligned = 0;
    StageAligned = 1;
  }
  optional StageType            kind      = 1;
  optional C843StageController  c843      = 2;
  optional SimulatedStage       simulated = 3;  
  optional TilingMode           tilemode  = 5;
}
